File: app/src/main/java/com/cx/retrofitdemo/bean/BaseBean.java
package com.cx.retrofitdemo.bean;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by chengxiao on 2018/6/5.
 */

public class BaseBean<T> {


    private int Code;
    private boolean Success = false;
    private String Errors = "";
    private ResultState resultState = ResultState.DEFAULT;
    private List<T> Data;
    //    private List<IndexPreferredDataBean> Data;


    public ResultState getResultState() {
        return resultState;
    }

    public BaseBean<T> setResultState(ResultState resultState) {
        this.resultState = resultState;
        return this;
    }


    public boolean isSuccess() {
        return Success;
    }

    public BaseBean<T> setSuccess(boolean success) {
        Success = success;
        return this;
    }

    public String getErrors() {
        return Errors == null ? "" : Errors;
    }

    public BaseBean<T> setErrors(String errors) {
        Errors = errors;
        return this;
    }

    public List<T> getData() {
        if (Data == null) {
            return new ArrayList<>();
        }
        return Data;
    }

    public BaseBean<T> setData(List<T> data) {
        Data = data;
        return this;
    }

    public int getCode() {
        return Code;
    }

    public void setCode(int Code) {
        this.Code = Code;
    }


    @Override
    public String toString() {
        return "BaseBean{" +
                "Code=" + Code +
                ", Success=" + Success +
                ", Errors='" + Errors + '\'' +
                ", Data=" + Data +
                '}';
    }
}

File: app/src/main/java/com/cx/retrofitdemo/bean/BaseBean.kt
package com.cx.retrofitdemo.bean
import com.google.gson.annotations.SerializedName


/**
 * @ClassName: BaseBean
 * @Author: CX
 * @Date: 2020/7/9 11:04
 */
data class BaseBean<T>(
    @SerializedName("Code")
    val code: Int = 0, // 2000
    @SerializedName("Data")
    val `data`: List<T> = listOf(),
    @SerializedName("Errors")
    val errors: String = "", // null
    @SerializedName("Msg")
    val msg: String = "", // 请求成功!
    @SerializedName("PM")
    val pM: PM = PM(),
    @SerializedName("Success")
    val success: Boolean = false, // true
    var resultState :ResultState= ResultState.DEFAULT
) {
    data class PM(
        @SerializedName("PageCount")
        val pageCount: Int = 0, // 8
        @SerializedName("PageIndex")
        val pageIndex: Int = 0, // 1
        @SerializedName("PageSize")
        val pageSize: Int = 0, // 10
        @SerializedName("TotalCount")
        val totalCount: Int = 0 // 79
    )
}